### ISSUE TITLE
Request to support dynamic.clear()

### ISSUE THREAD

**Issue by k-ye** [taichi-dev/taichi#371]
Hi Yuanming,

I could't find a similar feature request, so I'm filing one here. Either this or #332 will be nice to have...  

I think that some bits and pieces for this feature already exist:

* [`struct dynamic` supports `clear()`](https://github.com/taichi-dev/taichi/blob/1cc91f52acebb2bb0e39a49baecbb459b60389d7/taichi/struct.h#L722)
* [`tlang` can emit Clear stmt](https://github.com/taichi-dev/taichi/blob/bfd2d16e3de733967901eb72cbb41104951bd55f/taichi/tlang.h#L169)

Missing parts:

* [LLVM codegen doesn't handle `SNodeOpType::clear` yet](https://github.com/taichi-dev/taichi/blob/fab54a1ed3216493559d93600b4dfe14653d87af/taichi/backends/codegen_llvm.h#L853)
* [python binding not created](https://github.com/taichi-dev/taichi/blob/11bcb648095e126997e788f4e017d217af765954/taichi/python_bindings.cpp#L179)

I'd like to help contribute on this, but I don't have CUDA :( Do you think it's still possible for the development?

**Comment by yuanming-hu** [comment]
Hi Ye,

I guess #332 is easier to do. We do have `dynamic.clear` supported in the legacy backend, yet the new LLVM backend doesn't have this yet. Also, we need to think about the memory allocator before we can do this, since `clear` involves garbage collection.

If you would like to help with #332, please take a look at `ti.append`, which appends to a dynamic node and returns the original node length.
Example usage: https://github.com/taichi-dev/taichi/blob/master/tests/python/test_dynamic.py#L134  Definition: https://github.com/taichi-dev/taichi/blob/master/python/taichi/lang/ops.py#L190

**Comment by yuanming-hu** [comment]
@k-ye I updated the developer installation doc (https://taichi.readthedocs.io/en/latest/dev_install.html). Please follow that instead of `dev_setup.py`. You are probably the first guy following the new installation doc, so please feel free to let me know if you run into any issues. Thanks!

**Comment by yuanming-hu** [comment]
Btw, which OS are you using?

**Comment by k-ye** [comment]
Nice, let me follow the updated process then :)

I'm on `macOS Catalina 10.15.2` right now, and ran into problems like #357 ... Right now i'm just replacing `ti.GUI` with `pygame`, which seems good enough to run the examples :)

**Comment by yuanming-hu** [comment]
Oh nice. Let's move the discussion to https://github.com/taichi-dev/taichi/issues/313 then. Thanks for helping with this!

**Comment by yuanming-hu** [comment]
This is now implemented, yet not fully tested and optimized. I'm optimizing its performance now.

**Comment by k-ye** [comment]
Thanks! Could you link the relevant commits here? I thought there was gonna be a [`Dynamic_clear()`] in [this file](https://github.com/taichi-dev/taichi/blob/master/taichi/runtime/node_dynamic.h), but clearly this was not the case...

**Comment by yuanming-hu** [comment]
It's called `Dynamic_deactivate`: 

https://github.com/taichi-dev/taichi/blob/master/taichi/runtime/node_dynamic.h#L41

**Comment by yuanming-hu** [comment]
Usage: https://github.com/taichi-dev/taichi/blob/691f41837ef61f87d02f87a5020d9915542ff0f4/tests/python/test_sparse_deactivate.py#L104

(Still, more tests are needed...)

**Comment by k-ye** [comment]
Cool thanks!
